jfxtras-agenda/src/main/java/jfxtras/internal/scene/control/skin/agenda/base24hour/AgendaSkinTimeScale24HourAbstract.java

```diff
@@ -238,7 +241,15 @@ public void refresh() {
 		refreshLocale();
 		setupAppointments();
 		nowUpdateRunnable.run(); 
-	}
+	}
+
+	/**
+	 * 
+	 */
+	@Override
+    public Pane getNodeForPopup(Appointment appointment) { return appointmentNodeMap.get(appointment); }
+    final private Map<Appointment, Pane> appointmentNodeMap = new WeakHashMap<>();
```

Comment:
Yes, that is a problem.  I can think of three choices:
1.  Not override equals and hashcode.
2.  Override equals only and not override hashcode.
3.  Not uses a map for getNodeForPopup.

Both options present problems.
1.  I need equals for unit testing.  I will have to use my own Appointment implementation for testing again.
2.  Unknown problems due to not following expected conventions.
3.  Other matching method required which is less efficient than a map (e.g. scan through a list).

Can you think of other options?